onnxruntime/test/framework/data_types_test.cc

```diff
@@ -204,7 +204,12 @@ struct MapTypeProto {
 class DataTypeTest : public testing::Test {
  public:
   static void SetUpTestCase() {
-    RegisterTestTypes();
+    // xcTest run test case by case, so the SetUp needs to be reentrant.
+    static std::atomic<bool> loaded(false);
+    if (!loaded.load()) {
+      loaded.store(true);
+      RegisterTestTypes();
+    }
```

Comment:
Is it possible, if loaded is false, 2 threads can enter inside if (!loaded.load()){...} if timing is perfect? 
Do you need a lock_guard to make this if statement atomic?

Reply:
You are right, But ORT test cases are not compliance with thread-safe, so no need to worry about the MP in test code.